File: Public/Scripts/Content/Account/SelfUser.ts
import {User} from "../../Common/User.js";
import {Server} from "../../Common/Server.js";
import {Channel} from "../../Common/Channel.js";

export class SelfUserContent
{
    public user: User;
    public email: string;
    public nicknames: Map<Server, string>;
    public servers: Server[];
    public friends: User[];
    public friendRequests: User[];
    public directMessages: Channel[];

    constructor(
        user: User,
        email: string,
        nicknames: Map<Server, string>,
        servers: Server[],
        friends: User[],
        friendRequests: User[],
        directMessages: Channel[],
    ) {
        this.user = user;
        this.email = email;
        this.nicknames = nicknames;
        this.servers = servers;
        this.friends = friends;
        this.friendRequests = friendRequests;
        this.directMessages = directMessages;
    }
}

File: Internal/Content/Account/SelfUser.cs
using BonfireServer.Internal.Common;
using BonfireServer.Internal.Content.Server;

namespace BonfireServer.Internal.Content.Account;

public class SelfUser(User user) : IBaseContent
{
    public User User { get; set; } = user;
    public string Email { get; set; } = user.Email;
    public Dictionary<ServerInfo, string> Nicknames { get; set; } = user.Nicknames.ToDictionary(x => new ServerInfo(x.Key), x => x.Value);
    public List<ServerInfo> Servers { get; set; } = user.Servers.Select(x => new ServerInfo(x)).ToList();
    public List<User> Friends { get; set; } = user.Friends;
    public List<User> FriendRequests { get; set; } = user.FriendRequests;
    public List<Channel> DirectMessages { get; set; } = user.DirectMessages;
}